src/App.tsx

```diff
@@ -2,8 +1,51 @@
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+type Movie = {
+  imdbId: string;
+  title: string;
+  description: string;
+  imdbUrl: string;
+  imgUrl: string;
+};
+
+type FilterOptions = {
+  query: string;
+};
+
+function getFilteredMovies(movies: Movie[], { query }: FilterOptions): Movie[] {
+  let filteredMovies = [...movies];
+
+  const cleanedQuery = query.trim().toLowerCase();
+
+  if (!cleanedQuery) {
+    return movies;
+  }
+
+  if (query) {
+    filteredMovies = filteredMovies.filter(
+      movie =>
+        (movie.title && movie.title.toLowerCase().includes(cleanedQuery)) ||
+        (movie.description &&
+          movie.description
+            .toLowerCase()
+            .replace(' ', '')
+            .includes(cleanedQuery)),
+    );
+  }
```

Comment:
The condition `if (query)` is redundant here because you already check for `cleanedQuery` earlier. You can remove this `if` block and directly filter the movies.